bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -1022,15 +1031,19 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, scope
 
 	// as secret creation and scope.Config status patch are not atomic operations
 	// it is possible that secret creation happens but the config.Status patches are not applied
-	if err := r.Client.Create(ctx, secret); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
+	secret, err := r.getSecret(ctx, scope.Config.Namespace, scope.Config.Name)
```

Comment:
Does this logic need to be changed? It looks like what was there before should catch all corner cases

Reply:
Added this in response to the comment here - https://github.com/kubernetes-sigs/cluster-api/issues/3134#issuecomment-638096811

Reply:
Not sure if that's actually needed (cc @fabriziopandini). That logic just informs that the secret already exists and exits; which sounds we should still do regardless